torch/autograd/gradcheck.py

```diff
@@ -401,6 +459,17 @@ def not_reentrant_error(error_str=''):
         if out_is_complex and not reentrant_with_imag_grad_out:
             return not_reentrant_error(' (calculated using complex valued grad output)')
 
+        if check_batched_grad:
+            assert reentrant, ('Batched gradient checking makes the assumption that '
```

Comment:
nit I would mention that we expect gradcheck to have early exited before reaching this point when not re-entrant.